### ISSUE TITLE
Questions about the metrics used in paper

### ISSUE THREAD

**Issue by wangkiw** [yaodongyu/ProjNorm#3]
Hi Yaodong,

Thanks for your exciting work！ 

I am confused about the metrics $\rho$ and $r^2$. The subsection `Metric` of section `3. Experimental Results` says  :
> To evaluate performance, we compute the correlation between the predictions and the actual test accuracies across the OOD test datasets, using $R^2$ and rank correlation (Spearman’s $\rho$). 

What is the meaning of **the predictions and the actual test accuracies** here? In my understanding, we will have 90 `(ProjNorm value, Test Error)` pairs in total after evaluating the model on OOD test sets (CIFAR10&ResNet18). How to compute the $R^2$ score of these 90 pairs by  [sklearn.metrics.r2_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html)?

Best,
Qi-Wei

**Comment by ZitongYang** [comment]
Hi Qi-Wei,

Thanks for your interest in the paper.

> What is the meaning of the predictions and the actual test accuracies here? In my understanding, we will have 90 `(ProjNorm value, Test Error)`

You are right, the  "predictions" just means the `ProjNorm value`.

> How to compute the score of these 90 pairs by `sklearn.metrics.r2_score`?

I am not exactly sure about the specific package you cited. But here we just mean the standard simple linear regression. We used the `lm` function from `R`. It performs `Test Error` against `ProjectNorm value`. And report the usual R2 statistics (https://en.wikipedia.org/wiki/Coefficient_of_determination).

**Comment by wangkiw** [comment]
Hi Zitong,

Thanks for the detailed reply! Now I understand the $R^2$ here. Thanks for your exciting work again.

Best, 
Qi-Wei
